fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -72,35 +72,94 @@ public class FDBDirectoryWrapper implements AutoCloseable {
     private final Tuple key;
     @Nonnull
     private final LuceneAnalyzerWrapper analyzerWrapper;
-    @SuppressWarnings({"squid:S3077"}) // object is thread safe, so use of volatile to control instance creation is correct
-    private volatile IndexWriter writer;
-    @SuppressWarnings({"squid:S3077"}) // object is thread safe, so use of volatile to control instance creation is correct
-    private volatile DirectoryReader writerReader;
+    private volatile boolean useWriter = false;
```

Comment:
I believe I was able to verify that the previous behavior was maintained through this change. It would be beneficial, though, maybe later, to add a little more documentation to the class - esp. around the writer/writerReader and `useWriter`